src/bin/pg_autoctl/pgctl.c

```diff
@@ -1684,6 +1679,19 @@ pg_log_startup(const char *pgdata, int logLevel)
 					  pgLogFilePath);
 			return false;
 		}
+
+		/* our logFiles are regular files, skip . and .. and others */
+		if (logFileDirEntry->d_type == DT_UNKNOWN)
+		{
+			if (!S_ISREG(pgLogFileStat.st_mode))
+			{
+				continue;
+			}
+		} else if (logFileDirEntry->d_type != DT_REG)
```

Comment:
Nitpicking (style): please have else if on its own line.

More importantly, can you add a comment about why it's important to first try the d_type DT_UNKNOWN and the ST_ISREG macro and then check the d_type DT_REG? I'm curious, and then I think future us will be too ;-)

Reply:
Sure thing!

Reply:
Updated